lib/turbo/test_assertions.rb

```diff
@@ -79,5 +79,76 @@ def assert_no_turbo_stream(action:, target: nil, targets: nil)
       selector << %([targets="#{targets}"]) if targets
       assert_select selector, count: 0
     end
+
+    # Assert that the rendered fragment of HTML contains a `<turbo-frame>`
+    # element.
+    #
+    # ==== Options
+    #
+    # * <tt>:id</tt> [String] matches the element's <tt>[id]</tt> attribute
+    # * <tt>:loading</tt> [String] matches the element's <tt>[loading]</tt>
+    #   attribute
+    # * <tt>:src</tt> [String] matches the element's <tt>[src]</tt> attribute
+    # * <tt>:target</tt> [String] matches the element's <tt>[target]</tt>
+    #   attribute
+    # * <tt>:count</tt> [Integer] indicates how many turbo frames are expected.
+    #   Defaults to <tt>1</tt>.
+    #
+    #   Given the following HTML fragment:
+    #
+    #     <turbo-frame id="example" target="_top"></turbo-frame>
+    #
+    #   The following assertion would pass:
+    #
+    #     assert_turbo_frame id: "example", target: "_top"
+    #
+    # You can also pass a block make assertions about the contents of the
+    # element. Given the following HTML fragment:
+    #
+    #     <turbo-frame id="example">
+    #       <p>Hello!</p>
+    #     </turbo-frame>
+    #
+    #   The following assertion would pass:
+    #
+    #     assert_turbo_frame id: "example" do
+    #       assert_select "p", text: "Hello!"
+    #     end
+    #
+    def assert_turbo_frame(id:, loading: nil, src: nil, target: nil, count: 1, &block)
```

Comment:
Had you considered changing the `id:` keyword argument to an `id` positional argument so that the assertion's interface mimics the [turbo_frame_tag](https://github.com/hotwired/turbo-rails/blob/main/app/helpers/turbo/frames_helper.rb#L38) helper?

```suggestion
    def assert_turbo_frame(*ids, loading: nil, src: nil, target: nil, count: 1, &block)
    
      id = ids.first.respond_to?(:to_key) ? ActionView::RecordIdentifier.dom_id(*ids) : ids.join('_')
```